### PR TITLE
Docs helm values complete

### PR DESCRIPTION
**Pull request by harveyxia**
Depends on:

~~1. https://github.com/solo-io/skv2/pull/224~~
~~2. https://github.com/solo-io/gloo-mesh-enterprise/pull/299~~
~~3. https://github.com/solo-io/gloo-mesh-enterprise/pull/302~~

Note: the reference doc for OSS won't appear in `latest` docs until the next PR since this PR itself contains the reference doc
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1382

### CODE REVIEW

File: codegen/generate.go [hunk 1]
```diff
@@ -176,3 +187,36 @@ func makeAgentCrdsCommand() codegen.Command {
...
+	f, err := os.OpenFile(filename, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0777)
+	if err != nil {
+		return eris.Errorf("error generating Helm values reference doc: %v", err)
+	}
+
+	if _, err := f.Write([]byte(helmValuesDoc)); err != nil {
+		return eris.Errorf("error generating Helm values reference doc: %v", err)
+	}
+
+	// change back to original directory
```
Comment (joekelley) [inline codegen/generate.go:216]: wdyt about `defer`ring this immediately after changing the directory to begin with, just to ensure we're always back in the expected directory?

Comment (harveyxia) [reply codegen/generate.go:216]: good call

File: docs/docsgen/copy_helm_values_docs.go [hunk 2]
```diff
@@ -6,20 +6,30 @@ import (
...
+
+	enterpriseFileMapping = map[string]string{
+		"enterprise-networking/codegen/helm/enterprise_networking_helm_values_reference.md": "%s/%s/enterprise_networking.md",
+		"enterprise-networking/codegen/helm/enterprise_agent_helm_values_reference.md":      "%s/%s/enterprise_agent.md",
+	}
 
-	fileMapping = map[string]string{
-		enterpriseNetworkingHelmValueDocPath: "%s/%s/enterprise_networking.md",
-		enterpriseAgentHelmValueDocPath:      "%s/%s/enterprise_agent.md",
+	rbacWwebookFileMapping = map[string]string{
```
Comment (joekelley) [inline docs/docsgen/copy_helm_values_docs.go:31]: -> rbacWebhookFileMapping

File: docs/docsgen/copy_helm_values_docs.go [hunk 3]
```diff
@@ -81,11 +165,14 @@ func copyHelmValuesDocsFromEnterprise(client *github.Client, rootDir string) err
...
-	contents, _, _, err := client.Repositories.GetContents(context.Background(), org, repo, path, &github.RepositoryContentGetOptions{
+	contents, _, resp, err := client.Repositories.GetContents(context.Background(), org, repo, path, &github.RepositoryContentGetOptions{
 		Ref: tag,
 	})
-	if err != nil {
+	if err != nil && resp.StatusCode != 404 {
 		return eris.Errorf("error fetching Helm values doc: %v", err)
+	} else if resp.StatusCode == 404 {
+		contextutils.LoggerFrom(context.Background()).Warnf("Helm values doc for repo: %s and path: %s not found", repo, path)
+		return nil
```
Comment (joekelley) [inline docs/docsgen/copy_helm_values_docs.go:175]: Why wouldn't this be a failure case?

Comment (harveyxia) [reply docs/docsgen/copy_helm_values_docs.go]: I'll make it a failure case and special case v1.0.0, which we won't backport the docs to

**Review by joekelley — APPROVED** [review]

**Pull request merged**